src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -292,15 +308,19 @@ async def list_dataset_questions(
     return Questions(items=dataset.questions)
 
 
-@router.get("/datasets/{dataset_id}/metadata-properties", response_model=MetadataProperties)
-async def list_dataset_metadata_properties(
+@router.get("/me/datasets/{dataset_id}/metadata-properties", response_model=MetadataProperties)
```

Comment:
This change will affect the frontend part. We need to sync with @damianpumar once the PR is ready. We can eventually apply the change in this PR.

@damianpumar The idea regarding metadata property visibility is just to return those metadata properties accessible by the requested user. So your frontend logic won't change. (You will receive only the visible properties)

Reply:
Also the Python SDK needs to be modified to send the requests to the new endpoint adding the `/me`

Reply:
Done!